openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSPatientServiceImpl.java

```diff
@@ -245,6 +250,49 @@ private Patient updatePatient(Config config, Patient patient) {
         return updatedPatient;
     }
 
+    private Patient updatePatientIdentifiers(Config config, Patient patient) {
+        Validate.notNull(patient, "Patient cannot be null");
+        Validate.notEmpty(patient.getUuid(), "Patient Id cannot be empty");
+
+        IdentifierType identifierType = new IdentifierType();
+        identifierType.setUuid(getMotechPatientIdentifierTypeUuid(config));
+
+        List<Identifier> identifiersList = patientResource.getPatientIdentifierList(config, patient.getUuid());
+        Patient updatedPatient;
+        try {
+            //Patient Identifiers have to be update separately.
+            updatePatientIdentifiers(config, patient, identifiersList);
+            updatedPatient = getPatientByUuid(config, patient.getUuid());
+            eventRelay.sendEventMessage(new MotechEvent(EventKeys.UPDATED_PATIENT_IDENTIFIERS_SUBJECT, EventHelper.patientParameters(updatedPatient)));
+        } catch (HttpClientErrorException e) {
+            throw new OpenMRSException("Failed to update OpenMRS patient's identifiers for patient with id: " + patient.getUuid(), e);
+        }
+        return updatedPatient;
+    }
+
+    private void updatePatientIdentifiers(Config config, Patient patient, List<Identifier> fetchedIdentifierList) {
+        String fetchedIdentifierName;
+        String newIdentifierName;
+        Boolean isUpdate = false;
+
+        for (Identifier newIdentifier : patient.getIdentifiers()) {
+            for (Identifier fetchedIdentifier : fetchedIdentifierList) {
+                fetchedIdentifierName = fetchedIdentifier.getIdentifierType().getName();
+                newIdentifierName = newIdentifier.getIdentifierType().getName();
+
+                if (newIdentifierName.equals(fetchedIdentifierName)) {
+                    fetchedIdentifier.setIdentifier(newIdentifier.getIdentifier());
+                    patientResource.updatePatientIdentifier(config, patient.getUuid(), fetchedIdentifier);
+                    isUpdate = true;
+                }
+            }
+            if (!isUpdate) {
+                patientResource.updatePatientIdentifier(config, patient.getUuid(), newIdentifier);
```

Comment:
i do not understand how it works, because newIdentifier do not have uuid, so why this is sent in update request ? I think there should be another request for adding new identifier to the existing list.